manager/src/main/java/com/redhat/service/bridge/manager/connectors/ConnectorsServiceImpl.java

```diff
@@ -0,0 +1,176 @@
+
+        persistConnector(processor, newConnectorName, connectorPayload);
+
+        try {
+            Connector connector = callConnectorService(connectorType, connectorPayload, newConnectorName);
+        } catch (Exception e) {
+            throw new RuntimeException(e); // TODO-MC error handling
+        }
+
+        return Optional.of(newConnectorName);
+    }
+
+    private void updateKafkaTopicInConnectorPayload(JsonNode connectorPayload, String kafkaTopicName) {
+        // Assume the Connector action is always transformed to a send-to-kafka Action
+        String kafkaTopic = kafkaTopicName;
+        ObjectNode kafka = connectorPayload.with("kafka");
+        kafka.put("topic", kafkaTopic);
+    }
+
+    private void persistConnector(Processor processor, String newConnectorName, JsonNode connectorPayload) {
+        ConnectorEntity newConnectorEntity = new ConnectorEntity();
+
+        newConnectorEntity.setName(newConnectorName);
+        newConnectorEntity.setStatus(ConnectorStatus.REQUESTED);
+        newConnectorEntity.setSubmittedAt(ZonedDateTime.now());
+        newConnectorEntity.setPublishedAt(ZonedDateTime.now());
+        newConnectorEntity.setProcessor(processor);
+        newConnectorEntity.setDefinition(connectorPayload);
+
+        connectorDAO.persist(newConnectorEntity);
+    }
+
+    private JsonNode parseConnectorPayloadString(Map<String, String> actionParameters) {
+        try {
+            String connectorPayloadString = actionParameters.get("connectorPayload");
+            return mapper.readTree(connectorPayloadString);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e); // TODO-MC error handling
+        }
+    }
+
+    private Connector callConnectorService(String connectorType,
+            JsonNode connectorPayload,
+            String newConnectorName) {
+        Connector createConnectorRequest = new Connector();
+
+        ConnectorAllOfMetadata metadata = new ConnectorAllOfMetadata();
+        metadata.setName(newConnectorName);
+        metadata.setKafkaId("kafkaId-ignored"); // this is currently ignored in the Connectors API
+        createConnectorRequest.setMetadata(metadata);
+
+        AddonClusterTarget deploymentLocation = new AddonClusterTarget();
+        deploymentLocation.setKind("addon");
+        deploymentLocation.setClusterId(mcClusterId);
+        createConnectorRequest.setDeploymentLocation(deploymentLocation);
+
+        createConnectorRequest.setConnectorTypeId(connectorType);
+
+        createConnectorRequest.setConnectorSpec(connectorPayload);
+
+        KafkaConnectionSettings kafka = new KafkaConnectionSettings();
+        kafka.setBootstrapServer(kafkaBootstrapServer);
+        kafka.setClientId(serviceAccountId);
+        kafka.setClientSecret(serviceAccountSecret);
+        createConnectorRequest.setKafka(kafka);
+
+        try {
+            Connector connectorResult = connectorsApi.createConnector(createConnectorRequest);
```

Comment:
@lucamolteni I assume this call is async and that we have to poll for the status of the Connector after this initial request? The thing we need to be careful of here is that we are invoking a remote system whilst potentially holding DB transaction locks on our tables (you're in a method annotated `@Transactional`). If for whatever reason MC is unresponsive then we need to make sure we don't hold the transaction open for a long time.

Reply:
@robpblake It's async (managed-service style, not HTTP async style), so it should return pretty fast. 
I think that we set a timeout and it hits it, it'll throw an exception and the row won't be committed. 

Alternatively, we could commit the transaction and then call the WS outside it and if it fails we update the processor row with `ERROR` - that will surely be more robust. 

WDYT? Anyway I'm going to tackle those details in https://issues.redhat.com/browse/MGDOBR-142